lib/src/cupertino_settings_item.dart

```diff
@@ -194,8 +194,21 @@ class CupertinoSettingsItemState extends State<CupertinoSettingsItem> {
     return GestureDetector(
       behavior: HitTestBehavior.opaque,
       onTap: () {
-        if (widget.onPress != null && widget.enabled) {
-          widget.onPress();
+        if ((widget.onPress != null || widget.onToggle != null) &&
+            widget.enabled) {
+          setState(() {
+            pressed = true;
+          });
+
+          if (widget.onPress != null) {
+            widget.onPress();
+          }
+
+          Future.delayed(const Duration(milliseconds: 100), () {
```

Comment:
Did you add the delay because of the ripple effect?

Reply:
Yes, the first setState to indicate the button has been tapped, the second to revert the color to it’s original state.

Reply:
Oh okay 👍